runtime-modules/storage/src/data_directory.rs

```diff
@@ -238,45 +253,31 @@ decl_module! {
         /// Maximum objects allowed per inject_data_objects() transaction
         const MaxObjectsPerInjection: u32 = T::MaxObjectsPerInjection::get();
 
-        /// Adds the content to the system. Member id should match its origin. The created DataObject
+        /// Adds the content to the system. Requires root privileges. The created DataObject
         /// awaits liaison to accept or reject it.
         #[weight = 10_000_000] // TODO: adjust weight
         pub fn add_content(
```

Comment:
add_content can call into multi_add_content since its just doing the same thing for a single upload.
Also I wonder if we need two separate calls, since you can upload a single item with `multi_add_content()` by passing a Vec of size 1.

Reply:
the add content calls need some good Error types to return in DispatchResult for the various ways they can fail. When you add quota limits there will be more ways they can fail.

Reply:
> add_content can call into multi_add_content since its just doing the same thing for a single upload.
> Also I wonder if we need two separate calls, since you can upload a single item with `multi_add_content()` by passing a Vec of size 1.

https://github.com/Joystream/joystream/pull/2103/commits/ae0e77f2a511ab24bd7cc07e689d20b8bfa87d74 Done here

Reply:
> the add content calls need some good Error types to return in DispatchResult for the various ways they can fail. When you add quota limits there will be more ways they can fail.

Don`t sure if i understand what you mean under  `good Error types`. I see, there are 2 different checks in `ensure_content_is_valid` that return different kind of errors.

Reply:
`When you add quota limits there will be more ways they can fail.`

I`ll add an additional type(s) of errors for quota limits feature